docs/src/main/asciidoc/security-csrf-prevention.adoc

```diff
@@ -9,11 +9,25 @@ include::_attributes.adoc[]
 
 https://owasp.org/www-community/attacks/csrf[Cross-Site Request Forgery (CSRF)] is an attack that forces an end user to execute unwanted actions on a web application in which they are currently authenticated.
 
-Quarkus Security provides a CSRF prevention feature which implements a https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#double-submit-cookie[Double Submit Cookie technique]. This techninque requires that the CSRF token is never directly exposed to scripts executed on the client-side. In this extension, the CSRF token is:
+Quarkus Security provides a CSRF prevention feature which implements https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#double-submit-cookie[Double Submit Cookie] and [CSRF Request Header] techniques.
 
-* sent as `HTTPOnly` cookie to the client, and
+`Double Submit Cookie` techninque requires that the CSRF token is never directly exposed to scripts executed on the client-side. CSRF token is:
+
+* sent as `HTTPOnly`, optionally signed, cookie to the client, and
 * directly embedded in a hidden form input of server-side rendered forms, which are transmitted to and used by the client.
 
+`CSRF Request Header` technique can be used to support server-side rendered HTMX or client-side Single-page applications (SPA) which do not use HTML `form` tags.
+
+For the server-rendered pages which do not use an HTML `form` tag, the CSRF token is never directly exposed to scripts executed on the client-side. CSRF token is:
```

Comment:
Well, same remark here, which makes less sense I think than the previous time, because here of course scripts have to access the token to set it as a header, no?

Reply:
Sure, it is CSRF cookie which is referenced here